### ISSUE TITLE
Bug Report - New parameter needed to save Credit Card with PayPal PayFlow

### ISSUE THREAD

**Issue by kimwhite** [strangerstudios/paid-memberships-pro#1674]
**Describe the bug**
There are some new parameters that have been created to ensure that recurring payments processing through the PayFlow Gateway are compliant with the credit card company’s processing requirements

**Expected behavior**

**From PayFlow**
"Since your payments are using the “RECURRING” tag on them, you will need to add a parameter to the payment requests. You will need to pass the name-value pair of “CARDONFILE=CITR” (Customer Initiated Transaction, Recurring) on the initial payment that created the recurring profile. If these profiles are being billed automatically, you would not need to make any other changes. If you create any follow-up payment using a reference transaction you would need to pass “CARDONFILE=MITR” (Merchant Initiated Transaction, Recurring) to the payment request.

Adding this parameter to the payment requests will make sure that all compliance requirements are met, and will make sure you do not need to migrate your recurring payments from PayFlow."

https://developer.paypal.com/docs/payflow/integration-guide/card-on-file/

**Comment by diversityX** [comment]
Thank you. I think it would be good to add the "CARDONFILE=MITR" to all recurring transactions once, to guarantee compliance for previously billed subscriptions. The non-compliance fees are stiff.

**Comment by ideadude** [comment]
@diversityX you recommend adding the "MITR" option, but wouldn't the "CITR" option make more sense? With PMPro all subscription set ups are initiated at checkout by the customer.

Or are you wondering if we can retroactively update the old subscriptions? Are the fees assessed on recurring payments?

**Comment by diversityX** [comment]
The fees are being assessed on each recurring payment that were not marked with the initial CITR tag from MasterCard through our merchant company. We have been threatened with a fine as well. We would need to add the CITR on all new subscriptions, but still need to deal with the past ones without it. I am told Authorize.net already is in compliance with each new transaction.

We have hundreds of recurring subscriptions that did not get the initial CITR, so it appears the MITR needs passing on these recurs without the CITR to be in compliance. Here it is straight from Paypal PayFlow Pro’s tech folk:

Since your payments are using the “RECURRING” tag on them, you will need to add a parameter to the payment requests. You will need to pass the name-value pair of “CARDONFILE=CITR” (Customer Initiated Transaction, Recurring) on the initial payment that created the recurring profile. If these profiles are being billed automatically, you would not need to make any other changes. If you create any follow up payment using a reference transaction you would need to pass “CARDONFILE=MITR” (Merchant Initiated Transaction, Recurring) to the payment request.

**Comment by sc0ttkclark** [comment]
@diversityX how many active memberships do you have on your site? I'm working on a script solution to update the existing profiles with Payflow so it sees the proper CARDONFILE setting.

**Comment by diversityX** [comment]
735 active members
164 of those are recurring PayFlow Pro.

Thanks!

**Comment by sc0ttkclark** [comment]
After further discussion with PayPal, this is what they have informed us upon our shared investigation into our usage here.

TLDR: It turns out that we didn't need to include `CARDONFILE` based on our API usage, so it's likely you may have faced a processor-specific problem you may want to follow up with PayPal about.

> Because you're using the TRXTYPE=R, you do not need to pass any sort of CARDONFILE variable. Payflow automatically handles all COF details when merchants are using our Recurring Billing profiles. Even that "TRXTYPE=S" doesn't need a COF variable because our system automatically detects that you've created a profile based off of that sale transaction.
>
> So, no COF variables are needed for these transactions, Payflow already does this automatically with the product you're using. This means that the merchant (edit: who reported this issue on your GitHub) is probably using a processor that doesn't support any COF variables. This is a problem with the Processor rather than with Payflow, so the merchant would need to switch off of that processor to one that supports COF as listed on this page. 
>
> I hope that helps and do let me know if you have further questions on this.
>
> Sincerely,
> Aaron
> Merchant Technical Support Engineer
> PayPal
